### ISSUE TITLE
Plugin not compatible with Shortcodes ultimate

### ISSUE THREAD

**Issue by Zodiac1978** [epiphyt/embed-privacy#130]
### Bug/Problem

Maybe this is out of scope to fix here. Then this is just a heads up.

Embed Privacy does not work correctly with Shortcodes Ultimate (Google Maps):
https://de.wordpress.org/plugins/shortcodes-ultimate/
https://getshortcodes.com/docs/google-maps/

It does block the map, but after clicking the box the map does not load.

### Steps to reproduce

1. Install Shortcodes Ultimate
2. Install Embed Privacy
3. Add map via Shortcode: For example `[su_gmap address="Stonehenge" responsive="yes"]`
4. See map is blocked (without Map logo)
5. Unblocking does not work

### Version

1.4.8

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Zodiac1978** [comment]
Looks like there are two issues here:

1. The map/iframe is loaded, but with height 0.

Fix -> Apply some custom CSS (problem is that the height is a dynamic value):
```
.is-enabled .embed-privacy-content {
  height: 400px;
}
```

2. Google Maps is not detected (missing logo and autoloading does not work correctly)

Possible fix: -> Changing the regex a bit:
`google\.com\/maps(\/(d\/)?embed)|.*output=embed.*`

Because Shortcodes Ultimate uses this markup:
`<div class="su-gmap su-u-responsive-media-%s%s"><iframe width="%s" height="%s" src="//maps.google.com/maps?q=%s&amp;output=embed%s" title="%s"></iframe></div>`

**Comment by MatzeKitt** [comment]
Thank you for the deep investigation!

I will consider it to be fixed in the version _after_ the upcoming 1.5.0 (except I have a huge motivation on the the weekend).

**Comment by MatzeKitt** [comment]
I used a slightly different regex that should be more strict and thus not match false positives.

Also I’m not a fan of forcing certain dimensions and used the idea of the plugin to have a responsive container with aspect ratio of 16:9 to display the iframe.

**Comment by Zodiac1978** [comment]
> I used a slightly different regex that should be more strict and thus not match false positives.

Of course! It was a quick proof-of-concept. Stricter is even better! 👍

> Also I’m not a fan of forcing certain dimensions and used the idea of the plugin to have a responsive container with aspect ratio of 16:9 to display the iframe.

Yes, I mentioned this in my issue but had no time to further explore a general solution and just added a note, that the actual size is dependent on the media used.

There are now "Shorts" on YouTube in Portrait mode (and YouTube accepts all forms of aspect ratios), so what happens if I want to have a different aspect-ratio than 16:9?

**Comment by MatzeKitt** [comment]
> There are now "Shorts" on YouTube in Portrait mode (and YouTube accepts all forms of aspect ratios), so what happens if I want to have a different aspect-ratio than 16:9?
>
> It looks like your solution is still building a hardcoded aspect-ratio ...

This should be a non-issue here since it only affects Google Maps iframes, not YouTube videos or others. For them, Embed Privacy should return the original aspect ratio as given from the embed code itself.

**Comment by Zodiac1978** [comment]
I mixed up Maps and Youtube. But what do you mean with:

> and used the idea of the plugin to have a responsive container with aspect ratio of 16:9 to display the iframe.

I am confused why the map should have a 16:9 aspect ratio.

**Comment by MatzeKitt** [comment]
The plugin Shortcodes Ultimate displays the map like this. I assume because it’s the only way to get an iframe responsive without the requirement of JavaScript.

**Comment by Zodiac1978** [comment]
Looking at the [docs page](https://getshortcodes.com/docs/google-maps/) this seems to be only true if the responsive parameter is true. Otherwise, you can specify width/height like you want.
